src/main/java/org/example/services/JobRoleService.java

```diff
@@ -19,27 +19,59 @@ public JobRoleService(final JobRoleDao jobRoleDao,
                           final JobRoleMapper jobRoleMapper) {
         this.jobRoleDao = jobRoleDao;
         this.jobRoleMapper = jobRoleMapper;
-
     }
 
     public List<JobRoleResponse> getAllJobRoles(
-            final int page, final int pageSize) throws SQLException {
-        try {
-            final int offset = (page - 1) * pageSize;
-            return jobRoleMapper.mapJobRoleListToJobRoleResponseList(
-                    jobRoleDao.getOpenJobRoles(offset, pageSize));
-        } catch (SQLException e) {
-            throw new SQLException(
-                    "Error fetching job roles with pagination: page="
-                    + page + ", pageSize="
-                    + pageSize, e);
-        }
+            final int page, final int pageSize, final String fieldName,
+            final String orderBy) throws SQLException {
+        final int offset = (page - 1) * pageSize;
+        return jobRoleMapper.mapJobRoleListToJobRoleResponseList(
+                jobRoleDao.getOpenJobRoles(offset, pageSize,
+                        fieldName, orderBy));
+}
+
+    public String getCurrentFieldFilter(final String fieldName)
```

Comment:
I dont see the point in this Luke - you pass in the fieldName just to pass it back out ?